src/Cassandra/Metrics/Registries/NodeMetrics.cs

```diff
@@ -0,0 +1,108 @@
+//    limitations under the License.
+//
+
+using System;
+
+using Cassandra.Connections;
+using Cassandra.Metrics.Abstractions;
+
+namespace Cassandra.Metrics.Registries
+{
+    /// <inheritdoc />
+    internal class NodeMetrics : INodeMetrics
+    {
+        private readonly IDriverMetricsProvider _driverMetricsProvider;
+        private readonly string _context;
+
+        private IHostConnectionPool _hostConnectionPool = null;
+
+        public NodeMetrics(IDriverMetricsProvider driverMetricsProvider, MetricsOptions metricOptions, string context)
+        {
+            _driverMetricsProvider = driverMetricsProvider;
+            _context = context;
+            MetricsRegistry = new InternalMetricsRegistry<NodeMetric>(driverMetricsProvider, metricOptions.DisabledNodeMetrics);
+
+            InitializeMetrics();
+        }
+
+        public IDriverCounter SpeculativeExecutions { get; private set; }
+
+        public IDriverMeter BytesSent { get; private set; }
+
+        public IDriverMeter BytesReceived { get; private set; }
+
+        public IDriverTimer CqlMessages { get; private set; }
+
+        public IDriverGauge OpenConnections { get; private set; }
+
+        public IDriverGauge AvailableStreams { get; private set; }
+
+        public IDriverGauge InFlight { get; private set; }
+
+        public IRequestErrorMetrics Errors { get; private set; }
+
+        public IRetryPolicyMetrics Retries { get; private set; }
+
+        public IRetryPolicyMetrics Ignores { get; private set; }
+
+        /// <inheritdoc />
+        public IInternalMetricsRegistry<NodeMetric> MetricsRegistry { get; }
+
+        private void InitializeMetrics()
+        {
+            try
+            {
+                SpeculativeExecutions = MetricsRegistry.Counter(
+                    _context, NodeMetric.Counters.SpeculativeExecutions, DriverMeasurementUnit.Requests);
+                BytesSent = MetricsRegistry.Meter(_context, NodeMetric.Meters.BytesSent, DriverMeasurementUnit.Bytes);
+                BytesReceived = MetricsRegistry.Meter(_context, NodeMetric.Meters.BytesReceived, DriverMeasurementUnit.Bytes);
+                CqlMessages = MetricsRegistry.Timer(
+                    _context, NodeMetric.Timers.CqlMessages, DriverMeasurementUnit.Requests, DriverTimeUnit.Milliseconds);
+
+                Errors = new RequestErrorMetrics(MetricsRegistry, _context);
+                Retries = new RetryPolicyOnRetryMetrics(MetricsRegistry, _context);
+                Ignores = new RetryPolicyOnIgnoreMetrics(MetricsRegistry, _context);
+
+                OpenConnections = MetricsRegistry.Gauge(
+                    _context, NodeMetric.Gauges.OpenConnections, () => _hostConnectionPool?.OpenConnections, DriverMeasurementUnit.None);
+
+                AvailableStreams = MetricsRegistry.Gauge(
+                    _context, NodeMetric.Gauges.AvailableStreams, () => _hostConnectionPool?.AvailableStreams, DriverMeasurementUnit.None);
```

Comment:
Is this necessary?
I think it doesn't give more info than `InFlight` and `OpenConnections` and it requires significant overhead to maintain it (2 interlocked ops per execution).

Reply:
That's a good point, I thought this would be useful as the java driver also has it, maybe for scenarios of very high throughput where it might be nice to see how close the connection is to being saturated? I don't know how much impact these 2 interlocked ops have on each execution, I still haven't started coding and running some benchmarks.

Reply:
We can always choose not to add this metric and property now. We can always implement it later if there is demand for it, wdyt

Reply:
Yeah, I'm +1 on removing it now, we can add it later if needed.

About op cost, there's a good estimation here: http://ithare.com/infographics-operation-costs-in-cpu-clock-cycles/

Reply:
Thanks for link, interesting stuff 👍 
Deleted `CountableConcurrentStack` and `AvailableStreams` metric